### DISCUSSION TITLE
Implement branch protection rules

### DISCUSSION THREAD

**Discussion by leopaul36** [TOSIT-IO/TDP#55] [Ideas]
GitHub can help us set [protected branches](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches).

Unfortunately, this has to be set per repository.

Here is my proposal for `tdp-collection` (thought it applied to every repo):

![image](https://user-images.githubusercontent.com/12017342/195369417-f2f90555-f67c-4321-9024-ff3a704435ab.png)

![image](https://user-images.githubusercontent.com/12017342/195369492-94ec60de-dedd-4fd2-8b58-8621c9bbfea5.png)

**Comment by rpignolet** [comment]
I agree except for the "Required number of approvals before merging: 2" because, for repo like `tdp-lib`, `tdp-server` and maybe `tdp-ui` we do not have 2 reviewers, for example, for `tdp-lib` and `tdp-server`, only @gboutry and me can review the code and can make useful comments. We will be happy if other can review but it is too early to do it in these repositories.

**Reply by leopaul36** [reply]
We can definitely lower this for particular repositories

**Reply by rpignolet** [reply]
Ok then.
